managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1214,27 +1214,74 @@ public void asyncFindNewestMatching(FindPositionConstraint constraint, Predicate
     @Override
     public void asyncFindNewestMatching(FindPositionConstraint constraint, Predicate<Entry> condition,
             FindEntryCallback callback, Object ctx, boolean isFindFromLedger) {
+        asyncFindNewestMatching(constraint, condition, null, null, callback, ctx,
+                isFindFromLedger);
+    }
+
+
+    @Override
+    public void asyncFindNewestMatching(FindPositionConstraint constraint, Predicate<Entry> condition,
+                                        Position start, Position end, FindEntryCallback callback,
+                                        Object ctx, boolean isFindFromLedger) {
         OpFindNewest op;
-        PositionImpl startPosition = null;
+        PositionImpl startPosition;
```

Comment:
Could you please add more explanation to help understand how this could improve the performance in the Motivation, also this make the method hard to read. ?

Reply:
Updated the description and the code, PTAL